File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-2.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        
        int n = nums.length;
        int l = 0, r = 0;
        int maxLen = Integer.MAX_VALUE;
        int sum = 0;

        while(r < n)
        {
            sum += nums[r];

            while(l <= r && sum >= target)
            {
                int len = r-l+1;
                maxLen = Math.min(maxLen,len);
                sum -= nums[l];
                l++;
            }

            r++;
        }

        return maxLen == Integer.MAX_VALUE ? 0 : maxLen;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ans = []
        _map = {}
        n = len(nums)

        for i in range(n):
            diff = target - nums[i]
            if diff in _map:
                ans.append(_map[diff])
                ans.append(i)
                break
            _map[nums[i]] = i
        
        return ans